File: Recursion/RecursionJava/src/Arrays/BinarySearch.java
package Arrays;


public class BinarySearch {
    public static void main(String[] args) {
        int []arr = {1,2,3,4,5,6,7,8,9};
        System.out.println(search(arr,1,0,arr.length-1));
    }
    public static int search(int[]arr , int target , int start , int end){
        if (start > end){
            return -1;
        }
        int mid = start + (end - start)/2;
        if (arr[mid] == target){
            return mid;
        }else if (target < arr[mid]){
            return search(arr,target,start,mid-1);
        }else{
            return search(arr,target,mid+1,end);
        }
    }
}

File: Recursion/RecursionC++/binarySearch.cpp
#include<iostream>
using namespace std;


bool binarySearch(int arr[] , int key , int low ,int high){
	if(low > high){
		return false ;
	}
	int mid = (low + high)/2 ; 
	if(arr[mid] == key){
		cout <<"Position of "<<key<< " is " <<mid+1 <<endl;
		return true ;
	}
	else if(key < arr[mid]){
		return binarySearch(arr , key , low , mid-1 );
	}
	else if( key > arr[mid]){
		return binarySearch(arr , key , mid+1 , high);
	}

}

int main(){
	int n ;
	cin >> n ;
	int arr[n];
	for(int i = 0 ; i < n ;i++){
		cin >> arr[i];
	}
	int key ;
	cin>> key;
	cout << binarySearch(arr , key ,0 , n-1);


	return 0;
}